File: Baekjoon/Class3/04.11047 동전 0.py
import sys
input = sys.stdin.readline

n, k = map(int, input().split())
coins = reversed([int(input()) for _ in range(n)]) #동전 종류

coin_count = 0 #동전 개수
for coin in coins:
    if k == 0: break
    coin_count += k // coin
    k %= coin
print(coin_count)

File: Baekjoon/Class3/04.11047 동전 0.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

let [n, k] = input[0].split(" ").map(Number);

const coins = new Array(n) // 동전 종류
for(let i = 1; i <= n; i++)
    coins[n - i] = input[i]; // 내림차순

let coinCount = 0; // 동전 개수
for(const coin of coins) {
    if(k === 0) break;
    coinCount += Math.floor(k / coin);
    k %= coin;
}
console.log(coinCount);